engine/source_sym_engine.ml

```diff
@@ -188,8 +196,14 @@ let group_constructors type_env (acs, rows) : (constructor * matrix) list * matr
     |> Seq.map (fun (k, group) -> (k, matrix_of_group group))
     |> List.of_seq
   in
-  let wildcard_matrix = matrix_of_group wildcard_group in
-  (constructor_matrices, wildcard_matrix)
+  let exhausted_all_cases = BatHashtbl.to_list group_tbl
+                            |> List.for_all (fun (kst, group) -> width type_env kst = group.arity)
```

Comment:
For me the question is whether the width of the type (which you can get by checking the constructor of the first group, if a "first group" exists) is equal to the number of groups in the table.

Reply:
(If there is no first group, all rows in the matrix start with a wildcard pattern, then we do need a fallback node.)